data-upgrade-news/src/main/java/org/exoplatform/news/upgrade/targets/PublishedNewsDisplayedPropUpgrade.java

```diff
@@ -67,61 +71,69 @@ public PublishedNewsDisplayedPropUpgrade(InitParams initParams,
     this.container = container;
   }
 
-  public int getPublishedNewsDisplayedPropCount() {
-    return publishedNewsDisplayedPropCount;
+  public int getMigratedPublishedNewsCount() {
+    return migratedPublishedNewsCount;
   }
 
   @Override
   public void processUpgrade(String oldVersion, String newVersion) {
     long startupTime = System.currentTimeMillis();
     LOG.info("Start upgrade of published news displayed property");
-
-    boolean transactionStarted = false;
-    RequestLifeCycle.begin(container);
+    ExoContainerContext.setCurrentContainer(container);
+    List<String> newsTargetMetadatas = metadataService.getMetadatas(NewsTargetingService.METADATA_TYPE.getName(), 0).stream()
+        .map(newsTargetMetadata -> String.valueOf(newsTargetMetadata.getId()))
+        .collect(Collectors.toList());
+    List<MetadataItemEntity> newsTargetsMetadataItems = new ArrayList<>();
+    RequestLifeCycle.begin(entityManagerService);
```

Comment:
It would be better to externalize this block into a dedicated method IMO and use `@ExoTransactional` on it. (This way, we will not need to start a transaction explicitely on it)